### ISSUE TITLE
notify users when Xcode relaunch required due to Xcode upgrade

### ISSUE THREAD

**Issue by alanhamlett** [alcatraz/Alcatraz#440]
When a new version of Xcode is released with a new `DVTPlugInCompatibilityUUID` there is a window of time where only Alcatraz is loaded, until Alcatraz updates each plugin. Once the plugin has been updated by Alcatraz, it won't be loaded until Xcode is relaunched and the user selects `Load Bundles` when prompted by Xcode's `Unexpected code bundle` popup.

When Alcatraz updates a plugin that requires an Xcode relaunch, Alcatraz should notify the user to relaunch Xcode and select `Load Bundles` when prompted.

This can happen even if the plugin maintainer adds support for the new `DVTPlugInCompatibilityUUID` before Alcatraz does, because Alcatraz might have not updated the plugin to the latest version before the user upgrades Xcode.

**Comment by jurre** [comment]
Yeah, this sounds like a great addition, how would we prompt the user though? Ideally we would not block Xcode with a regular popup

**Comment by alanhamlett** [comment]
We can create a [NSUserNotification](https://developer.apple.com/library/mac/documentation/Foundation/Reference/NSUserNotification_Class/index.html).

Just have the Alcatraz interface conform to the `NSUserNotificationCenterDelegate` protocol, create a `shouldPresentNotification` method that returns `YES`, then create a new `NSUserNotification` and display it by passing it to `deliverNotification` method on the `NSUserNotificationCenter`.

Some examples:

1. [NSUserNotification StackOverflow example](http://stackoverflow.com/questions/11814903/send-notification-to-mountain-lion-notification-center)
2. [NSUserNotification from an Xcode plugin example](https://github.com/alloy/AxeMode/blob/master/AxeMode/AxeMode.m)

**Comment by jurre** [comment]
Good idea! We used to have that in ATZ when installing a plugin: https://github.com/alcatraz/Alcatraz/pull/45/commits/c6c9d7aa3bd121379b3031829b4b353d972cd89b

**Comment by alanhamlett** [comment]
Cool! So it seems the things needed in a PR would be:

1. Before updating a package, save it's `DVTPlugInCompatibilityUUID`s.
2. After a package is updated, check it's `DVTPlugInCompatibilityUUID`s against the current Xcode UUID.
3. If the current Xcode UUID was not found before updating, but is found after updating then add plugin to a queue.
4. After updating all packages in `ATZPluginWindowController.packages` display a notification if plugins are in the queue.

Also, is [reloadXcodeForPackage](https://github.com/alcatraz/Alcatraz/blob/778193b76ea0b78a3fba607e87ea3769868b1e39/Alcatraz/Installers/ATZPluginInstaller.m#L89) related?
